ee/tabby-ui/app/(dashboard)/settings/general/components/security-form.tsx

```diff
@@ -0,0 +1,194 @@
+
+interface SecurityFormProps {
+  defaultValues?: Omit<Partial<SecurityFormValues>, 'domainList'> & {
+    domainList?: string[]
+  }
+  onSuccess?: () => void
+}
+
+export const GeneralSecurityForm: React.FC<SecurityFormProps> = ({
+  onSuccess,
+  defaultValues: propsDefaultValues
+}) => {
+  const defaultValues = React.useMemo(() => {
+    const _defaultValues: SecurityFormProps['defaultValues'] =
+      propsDefaultValues ?? {}
+    const { domainList, ...values } = _defaultValues
+    return {
+      disableTelemetry: false,
+      ...values,
+      domainList: buildListFieldFromValues(domainList) ?? [{ value: '' }]
+    }
+  }, [propsDefaultValues])
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues
+  })
+
+  const { fields, append, remove, update } = useFieldArray({
+    control: form.control,
+    name: 'domainList'
+  })
+
+  const isDirty = !isEmpty(form.formState.dirtyFields)
+
+  const onRemoveDomainItem = (index: number) => {
+    if (fields?.length === 1 && index === 0) {
+      update(index, { value: '' })
+    } else {
+      remove(index)
+    }
+  }
+
+  const handleDomainListKeyDown: React.KeyboardEventHandler<
+    HTMLInputElement
+  > = e => {
+    if (e.key === 'Enter' && !e.nativeEvent.isComposing) {
+      e.preventDefault()
+      append({ value: '' })
+    }
+  }
+
+  const onSubmit = ({ domainList, ...values }: SecurityFormValues) => {
+    const formattedValues = {
+      // format domainList before submit, Array<{ value: string }> => Array<string>
+      domainList: buildListValuesFromField(domainList),
+      ...values
+    }
+    // todo submit values
+  }
+
+  return (
+    <Form {...form}>
+      <div className="flex flex-col gap-4">
+        <form
+          className="flex flex-col gap-8"
+          onSubmit={form.handleSubmit(onSubmit)}
+        >
+          <FormField
+            control={form.control}
+            name="disableTelemetry"
+            render={({ field }) => (
+              <FormItem>
+                <div className="flex items-center gap-1">
+                  <FormControl>
+                    <Checkbox
+                      checked={field.value}
+                      onCheckedChange={field.onChange}
+                    />
+                  </FormControl>
+                  <FormLabel className="cursor-pointer">
+                    Disabling Client Side Telemetry
+                  </FormLabel>
+                </div>
+                <FormDescription>
+                  When activated, the client-side telemetry (IDE/Extensions)
+                  will be disabled, regardless of the client-side settings.
+                </FormDescription>
+              </FormItem>
+            )}
+          />
+          <div>
+            {fields.map((field, index) => (
+              <FormField
+                control={form.control}
+                key={field.id}
+                name={`domainList.${index}.value`}
```

Comment:
@wsxiaoys  Due to the implementation of 'useFieldArray', there is a data structure transformation here, and the form path will look like 'domainList.0.value'